cpp/include/cugraph_c/algorithms.h

```diff
@@ -392,6 +392,93 @@ cugraph_type_erased_device_array_view_t* cugraph_random_walk_result_get_path_siz
  */
 void cugraph_random_walk_result_free(cugraph_random_walk_result_t* result);
 
+/**
+ * @brief     Opaque neighborhood sampling result type
+ */
+typedef struct {
+  int32_t align_;
+} cugraph_sample_result_t;
+
+/**
+ * @brief     Uniform Neighborhood Sampling
+ *
+ * @param [in]  handle       Handle for accessing resources
+ * @param [in]  graph        Pointer to graph.  NOTE: Graph might be modified if the storage
+ *                           needs to be transposed
+ * @param [in]  start        Device array of start vertices for the sampling
+ * @param [in]  start_label  Device array of start labels.  These labels will propagate to the
+ * results so that the result can be properly organized when the input needs to be sent back to
+ * different callers (different processes or different gpus).
+ * @param [in]  fanout       Host array defining the fan out at each step in the sampling algorithm
+ * @param [in]  without_replacement
+ *                           Boolean value.  If true selection of edges is done without
+ *                           replacement.  If false selection is done with replacement.
+ * @param [in]  result       Output from the uniform_nbr_sample call
+ * @param [out] error        Pointer to an error object storing details of any error.  Will
+ *                           be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t uniform_nbr_sample(const cugraph_resource_handle_t* handle,
+                                        cugraph_graph_t* graph,
+                                        const cugraph_type_erased_device_array_view_t* start,
+                                        const cugraph_type_erased_device_array_view_t* start_label,
+                                        const cugraph_type_erased_host_array_view_t* fan_out,
+                                        bool_t without_replacement,
```

Comment:
No `do_expensive_check` here (at least as a placeholder even if we're not checking anything at this moment)?

Reply:
Added in latest push.